### DISCUSSION TITLE
Github.ref and failure() cannot be applied together?

### DISCUSSION THREAD

**Discussion by chenrui333** [community/community#27221] [Actions]
<p>I was trying to add some slack notifications into my github action job, but found that <code>gihtub.ref</code> check cannot be used with <code>success()</code>.</p>
<p>The code looks like below and both success() and failure() got executed all the time.</p>
<pre><code class="lang-auto">  job:
    ...
    steps:
      ...
      - name: success run notification
        if: github.ref == 'refs/heads/master' &amp;&amp; ${{ success() }}
        env:
          SLACK_WEBHOOK: ${{ secrets.LAUNCHING_PD_SLACK_WEBHOOK }}
        uses: Ilshidur/action-slack@2.1.0
        with:
          args: |
            success run!
      - name: failure run notification
        if: github.ref == 'refs/heads/master' &amp;&amp; ${{ failure() }}
        env:
          SLACK_WEBHOOK: ${{ secrets.LAUNCHING_PD_SLACK_WEBHOOK }}
        uses: Ilshidur/action-slack@2.1.0
        with:
          args: |
            failure run
</code></pre>
<p>The workaround I did is to move those two slack notification steps into separate job, so that I can do something like this, and it seems working for my favor.</p>
<pre><code class="lang-auto">  slack-notif:
    runs-on: ubuntu-latest
    needs: job
    if: github.ref == 'refs/heads/master'
    steps:
      - name: success run notification
        if: ${{ success() }}
        env:
          SLACK_WEBHOOK: ${{ secrets.LAUNCHING_PD_SLACK_WEBHOOK }}
        uses: Ilshidur/action-slack@2.1.0
        with:
          args: |
            success run!
      - name: failure run notification
        if: ${{ failure() }}
        env:
          SLACK_WEBHOOK: ${{ secrets.LAUNCHING_PD_SLACK_WEBHOOK }}
        uses: Ilshidur/action-slack@2.1.0
        with:
          args: |
            failure run!
</code></pre>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="chenrui333" data-post="1" data-topic="147303">
<div class="title">
<div class="quote-controls"></div>
 chenrui333:</div>
<blockquote>
<pre><code class="lang-yaml">        if: github.ref == 'refs/heads/master' &amp;&amp; ${{ success() }}
</code></pre>
</blockquote>
</aside>
<p>You need to wrap the whole condition expression in <code>${{ ... }}</code>, or omit it altogether (see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsif"><code>jobs.&lt;job_id&gt;.steps.if</code></a>):</p>
<pre><code class="lang-yaml">        if: ${{ github.ref == 'refs/heads/master' &amp;&amp; success() }}
</code></pre>
<p>Same for the <code>failure()</code> version.</p>

**Comment by chenrui333** [comment]
<p>Thanks, it works for me.</p>
